### PR TITLE
docs(data exchange): adapt and restructure the data exchange

### PR DESCRIPTION
**Pull request by stephanbcbauer**
## Description

This PR adds some improvements, restructuring within the data exchange documentation. It also adds requests and expected responses to get a more in-depth understanding of the flow. The currently documented flow was tested multiple times.

@dvg36 would be cool to test it again, just to be sure!!! I think after this, we are well-prepared for the Tractus-X Community Days. ;)

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [X] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [X] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by dvg36** [comment]
@stephanbcbauer I will test the tutorial tomorrow morning again.

File: docs/user/guides/data-exchange/provide-data.md [hunk 1]
```diff
@@ -1,112 +1,642 @@
...
+    "tx-auth": "https://w3id.org/tractusx/auth/",
+    "cx-policy": "https://w3id.org/catenax/policy/",
+    "odrl": "http://www.w3.org/ns/odrl/2/"
   }
-}'
+}
 ```
 
-## 2. Get Asset by ID
+The second request will return all assets. It`s called `request catalog` and will be explained in the next step.
```
Comment (mgarciaLKS) [inline docs/user/guides/data-exchange/provide-data.md:134]:
```suggestion
The second request will return all assets. It's called `request catalog` and will be explained in the next step.
```

File: docs/user/guides/data-exchange/provide-data.md [hunk 2]
```diff
@@ -1,112 +1,642 @@
...
+The second request will return all assets. It's called `request catalog` and will be explained in the next step.
 
-To check if the asset is correctly created.
+## Request the catalog
 
-```
-curl -L -X GET 'http://dataprovider-controlplane.tx.test/management/v3/assets/200' \
--H 'X-Api-Key: TEST2'
-```
+Once the asset is created, Alice (the data consumer) can try request a catalog of available (offered) data.
```
Comment (mgarciaLKS) [inline docs/user/guides/data-exchange/provide-data.md:138]:
```suggestion
Once the asset is created, Alice (the data consumer) can try to request a catalog of available (offered) data.
```

**Review by evegufy — COMMENTED** [review]
LGTM!
Could you please also fix this [dead link](https://github.com/eclipse-tractusx/tractus-x-umbrella/blob/main/docs/user/setup/README.md?plain=1#L70)?

File: docs/user/guides/data-exchange/provide-data.md [hunk 3]
```diff
@@ -1,112 +1,642 @@
...
+    "odrl": "http://www.w3.org/ns/odrl/2/"
   }
-}'
+}
 ```
 
-## 4. Get Policies by ID
+### Step 2: Request catalog - second try
+
+Now that Bob created an access policy, Alice can once again try to access Bob's offer by running this `curl`.
```
Comment (mgarciaLKS) [inline docs/user/guides/data-exchange/provide-data.md:360]:
```suggestion
Now that Bob has created an access policy, Alice can once again try to access Bob's offer by running this `curl`.
```

File: docs/user/guides/data-exchange/consume-data.md [hunk 4]
```diff
@@ -1,127 +1,273 @@
...
-					"odrl:rightOperand": "BPNL00000003AZQP"
-				}
-			}
-		},
-		"prohibition": [],
-		"obligation": [],
-		"target": "200"
-	}
-}'
+We learned already about the response of a catalog request in the [Provide Data](provide-data.md) guide. The response contains a list of assets that Alice can consume from Bob.
```
Comment (mgarciaLKS) [inline docs/user/guides/data-exchange/consume-data.md:37]:
```suggestion
We have already learned about the response of a catalog request in the [Provide Data](provide-data.md) guide. The response contains a list of assets that Alice can consume from Bob.
```

File: docs/user/guides/data-exchange/consume-data.md [hunk 5]
```diff
@@ -1,127 +1,273 @@
...
+  }
+}
 ```
 
-You should be able to see in the response that the _state_ value is equal to _STARTED_.
+## Step 5: Fetch data
 
-## 6. Validate Transfer
+Using the {{ENDPOINT}} and {{TOKEN}} from the response, Alice fetches the data. In our example the `uuid` `urn:uuid:b77c6d51-cd1f-4c9d-b5d4-091b22dd306b` for the data is already known. In
+real life you would receive this information from the digital twin registry asn an endpoint in the submodelDescriptors.
```
Comment (mgarciaLKS) [inline docs/user/guides/data-exchange/consume-data.md:223]:
```suggestion
real life you would receive this information from the digital twin registry as an endpoint in the submodelDescriptors.
```

**Review by mgarciaLKS — APPROVED** [review]
I think it's a very good idea to add Alice and Bob to this great story. Again, very nice job rewriting the documentation, in this case of the data exchange.

I have followed the “tutorial” locally and everything works correctly. I have only found small details in the writing, that with all that you have written, I am surprised there are only those (that I have encountered).

For my part, everything works very well and the PR is great, and all the work performed is going to be valued.

**Comment by stephanbcbauer** [comment]
> I think it's a very good idea to add Alice and Bob to this great story. Again, very nice job rewriting the documentation, in this case of the data exchange.
> 
> I have followed the “tutorial” locally and everything works correctly. I have only found small details in the writing, that with all that you have written, I am surprised there are only those (that I have encountered).
> 
> For my part, everything works very well and the PR is great, and all the work performed is going to be valued.

Thank you very much @mgarciaLKS for your effort and detailed testing, :) I applied your suggestions. Hopefully, it worked.

**Comment by stephanbcbauer** [comment]
> LGTM! Could you please also fix this [dead link](https://github.com/eclipse-tractusx/tractus-x-umbrella/blob/main/docs/user/setup/README.md?plain=1#L70)?

done thx

**Review by evegufy — APPROVED** [review]

**Review by mgarciaLKS — APPROVED** [review]

**Pull request merged**